### DISCUSSION TITLE
Could you please provide more information about the threshold formulas?

### DISCUSSION THREAD

**Discussion by criker1** [tronikos/esphome-magnetometer-water-gas-meter#17] [General]
My sensor is picking up the following signal during a rotation:

![grafik](https://github.com/tronikos/esphome-magnetometer-water-gas-meter/assets/55559981/638928e4-fce4-43b9-8eba-5f15ad1b76b8)

The maximum and minimum values would be:
max -> 273 µT
min -> -273 µT

With the threshold formulas provided, I would get:
Threshold lower = min + 0.25 * (max - min) -> -273 + 0.25 * (273 - (-273)) µT = -273 + 136.5 µT = -136.5 µT
Threshold upper = max - 0.25 * (max - min) -> 273 - 0.25 * (273 - (-273)) µT= 273 - 136.5 µT = +136.5 µT

Why should I use these calculated values instead of +200 µT and -200 µT?

I am asking because the spike in the red square already exceeds 136.5 µT.

Just for my understanding, could you give any information on where the threshold formulas are coming from?

Many thanks for your answer and the great work you did!

**Comment by tronikos** [comment]
A rotation is detected when values cross the thresholds. If there was no signal noise, one threshold at the center point would be enough. But because of noise the code selects as thresholds  of +-5 from the center point. I haven't update the README which says +-50% of peak. That doesn't work well for high flow meters because of the 5ms sampling frequency.
